SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -627,35 +710,209 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
 
 			if (Actor != nullptr)
 			{
-				FVector ActorLocation = Actor->GetActorLocation();
-
-				if (ActorLocation.IsZero())
+				FVector2D ScreenLocation = ProjectActorToScreen(Actor, PlayerLocation);
+				if (ScreenLocation.IsZero())
 				{
 					continue;
 				}
 
-				if (FVector::Dist(PlayerLocation, ActorLocation) > MaxRange)
+				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/);
+			}
+		}
+	}
+}
+
+void ASpatialDebugger::SelectActorsToTag(UCanvas* Canvas)
```

Comment:
This does both actor selection and tag drawing, would be great if the name reflected that. Struggling to find a concise name though :/